### ISSUE TITLE
New property `maxLines` in ReturnCountCheck to allow multiple `return` in small methods

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#16298]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/returncount.html
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```
D:\CS\test cat src/Test.java
public class Test {
    public int shortMethod(int x) {       // violation
        if (x % 2 == 0) {
            return x / 2;
        }
        return x * 2;
    }

    public int longMethod(int x) {    // violation
        if (x < 0) {
            return -1;  
        }

        int sum = 0;

        for (int i = 0; i < x; i++) {
            sum += i;
            if (i > 100) {
                return 100;
            }
        }

        if (x > 100) {
            return 100;
        }

        return sum;
    }
}
D:\CS\test cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="severity" value="error"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="max" value="1"/>
        </module>
    </module>
</module>
D:\CS\test
java  -jar checkstyle-10.21.1-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:2:5: Return count is 2 (max allowed for non-void methods/lambdas is 1). [ReturnCount]
[ERROR] D:\CS\test\src\Test.java:9:5: Return count is 4 (max allowed for non-void methods/lambdas is 1). [ReturnCount]
Audit done.
Checkstyle ends with 2 errors.
```

---

**Is your feature request related to a problem? Please describe.**

The single-exit rule provides benefits primarily in larger functions by improving maintainability and control flow. However, in small functions, enforcing this rule offers little to no benefit and can even reduce readability.

For example, the method `shortMethod` in the previous example does not harm readability, and flagging it as a violation would be unnecessary.

To support this argument, I found a relevant paragraph from _Clean Code_ by Robert C. Martin:

> Some programmers follow Edsger Dijkstra’s rules of structured programming. Dijkstra said that every function, and every block within a function, 
> should have one entry and one exit. Following these rules means that there should only be one return statement in a function

> Those rules serve little benefit when functions are very small. It is only in larger functions
>  that such rules provide significant benefits. So if you keep your functions small, then the occasional multiple return, break, or
> continue statement does no harm and can sometimes even be more expressive than the single-entry, single-exit rule.

---

**Describe the solution you'd like**
A new property called `maxLines`, This property would define the maximum number of lines a function can have before multiple return statements are flagged. If a function has fewer lines than the specified threshold, multiple returns are allowed.

| name  | description |  Default Value |
| -------------  | ------------- | ---------- |
| maxLines  | Specify the maximum number of lines a function can have before multiple return statements are flagged | 10 |

```bash
java  -jar checkstyle-10.21.1-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:9:5: Return count is 4 (max allowed for non-void methods/lambdas is 1). [ReturnCount]
Audit done.
Checkstyle ends with 1 errors.
```

`ShortMethod` in the above example has less than 10 lines so if will be skipped, The second method has more than 10 so if will be checked for violation
